src/hooks/useChainId.ts

```diff
@@ -44,7 +46,15 @@ export const useUrlChainId = (): string | undefined => {
 export const useChainId = (): string => {
   const session = useAppSelector(selectSession)
   const urlChainId = useUrlChainId()
-  return urlChainId || session.lastChainId || defaultChainId
+  const wallet = useWallet()
+  const chains = useChains()
+
+  const fallbackChainId =
+    wallet?.chainId && chains?.configs.some(({ chainId }) => chainId === wallet.chainId)
+      ? wallet.chainId
+      : defaultChainId
+
+  return urlChainId || session.lastChainId || fallbackChainId
```

Comment:
I would define the wallet chainId as a separate thing and fall back in this order:
```
  return urlChainId || walletChainId || session.lastChainId || fallbackChainId
```

Reply:
Is `fallbackChainId` and `walletChainId` the same thing?

Reply:
I've inverted the order and added tests to cover it.

Reply:
> Is fallbackChainId and walletChainId the same thing?

In my previous commit, `fallbackChainId` was either that of the `wallet` or from `defaultChainId`. I've changed it in accordance with the first suggestion. Wdyt?